controllers/scaledobject_controller.go

```diff
@@ -214,6 +219,29 @@ func (r *ScaledObjectReconciler) ensureScaledObjectLabel(logger logr.Logger, sca
 	return r.Client.Update(context.TODO(), scaledObject)
 }
 
+func (r *ScaledObjectReconciler) validateMetricNameUniqueness(logger logr.Logger, scaledObject *kedav1alpha1.ScaledObject) error {
+	scalers, err := r.scaleHandler.GetScalers(scaledObject)
+	if err != nil {
+		return err
+	}
+
+	observedMetricNames := make(map[string]struct{})
+	for _, scaler := range scalers {
+		for _, metric := range scaler.GetMetricSpecForScaling() {
+			metricName := metric.External.Metric.Name
```

Comment:
We should check whether the scaler contains External metrics (`if != nil`), becasue if Resource based Metrics are used, External Metrics are not set.

See: https://github.com/kedacore/keda/blob/af2b36f593c231ed25c48a0ba432f6128f77cbf5/controllers/hpa.go#L135-L162

Reply:
That's a great point! I went ahead and added a continue when external metrics is nil